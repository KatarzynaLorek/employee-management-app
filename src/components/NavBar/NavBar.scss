.navbar {
  position: fixed;
  top: 0;
  left: 0;
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 200px;
  height: 100vh;
  padding-top: 50px;
  background-color: #77b28c;
  box-shadow: 4px 4px 8px -1px rgba(0, 0, 0, 0.18);
  transition: background-color 0.2s linear, color 0.2s linear;
  z-index: 9999;

  &__title {
    height: 150px;
    font-weight: 700;
    line-height: 30px;
    font-size: 2.5rem;
    text-transform: uppercase;
    text-align: center;
    color: #000;
    text-decoration: none;
    text-transform: uppercase;
  }

  &__navigation {
    display: flex;
    flex-direction: column;
    align-items: center;
  }

  &__button {
    margin-top: 10px;
    height: 50px;
    width: 150px;
    border: 0;
    border-radius: 20px;
    background-color: #77b28c;
    color: #000;
    font-size: 1.6rem;
    line-height: 50px;
    text-align: center;
    text-decoration: none;
    text-transform: uppercase;
    cursor: pointer;

    &--active {
      font-weight: 700;
      background-color: #fff;
    }
  }
}

@media (max-width: 1250px) {
  .navbar {
    width: 100%;
    height: 100px;
    padding-top: 10px;

    &__title {
      width: 500px;
    }

    &__navigation {
      flex-direction: row;
      justify-content: center;
      padding-bottom: 50px;
    }

    &__button {
      height: 40px;
      line-height: 40px;
      margin-left: 5px;
      margin-right: 5px;
    }
  }
}

@media (max-width: 411px) {
  .navbar {
    &__title {
      width: 360px;
      font-size: 2.3rem;
    }
  }
}

@media (max-width: 360px) {
  .navbar {
    &__title {
      width: 360px;
      font-size: 2.2rem;
    }
  }
}

@media (max-width: 320px) {
  .navbar {
    &__title {
      width: 320px;
      font-size: 2rem;
    }
  }
}
