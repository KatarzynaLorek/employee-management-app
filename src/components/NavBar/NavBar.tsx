import React from 'react';
import { NavLink } from 'react-router-dom';
import './NavBar.scss';

const NavBar: React.FC = () => {
  return (
    <>
      <div className="navbar">
        <NavLink exact to="/" className="navbar__title">
          Employee management app
        </NavLink>
        <nav className="navbar__navigation">
          <NavLink exact to="/" className="navbar__button" activeClassName="navbar__button--active">
            Table
          </NavLink>
          <NavLink
            exact
            to="/charts"
            className="navbar__button"
            activeClassName=" navbar__button--active"
          >
            Charts
          </NavLink>
        </nav>
      </div>
    </>
  );
};

export default NavBar;
