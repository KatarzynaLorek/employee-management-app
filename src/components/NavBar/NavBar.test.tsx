import React from 'react';
import { Router } from 'react-router-dom';
import { render } from '@testing-library/react';
import { createMemoryHistory } from 'history';
import NavBar from './NavBar';

const renderNavBar = () => {
  const history = createMemoryHistory();
  return render(
    <Router history={history}>
      <NavBar />
    </Router>,
  );
};

describe('NavBar component', () => {
  it('renders NavBar properly', () => {
    const { getByText } = renderNavBar();
    expect(getByText('Employee management app')).toBeInTheDocument();
  });
});
