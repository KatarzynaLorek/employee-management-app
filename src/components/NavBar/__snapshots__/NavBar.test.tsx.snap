// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar component renders NavBar properly 1`] = `
<body>
  <div>
    <div
      class="navbar"
    >
      <a
        aria-current="page"
        class="navbar__title active"
        href="/"
      >
        Employee management app
      </a>
      <nav
        class="navbar__navigation"
      >
        <a
          aria-current="page"
          class="navbar__button navbar__button--active"
          href="/"
        >
          Table
        </a>
        <a
          class="navbar__button"
          href="/charts"
        >
          Charts
        </a>
      </nav>
    </div>
  </div>
</body>
`;
